src/Types.hs

```diff
@@ -88,38 +139,63 @@ data Category =
   deriving (Eq, Ord, Bounded, Enum, Show, Read, Generic)
 
 -- | "Verbesserungsvorschlag"
+--
+-- 'Comments' are hierarchical.  The application logic is responsible for putting some limit on the
+-- recursion depth under which all children become siblings.  (FIXME: or should we represent this in
+-- the types?)
 data Comment = Comment
-    { _commentMeta    :: MetaInfo
-    , _commentArticle :: Article
-    , _commentVotes   :: Set Vote
+    { _commentMeta    :: MetaInfo Comment
+    , _commentText    :: Document
+    , _commentVotes   :: Set Comment
+    , _commentReplies :: Set CommentVote
     }
   deriving (Eq, Ord, Show, Read, Generic)
 
--- | "Stimme"
-data Vote = Vote
-    { _voteMeta  :: MetaInfo
-    , _voteValue :: Maybe Bool
+-- | "Stimme" for "Idee"
```

Comment:
Right, I missed that spot.  Fixed.

This raises the question of whether we should have German in the source code?  I did it because it references the specs we have in German, but if anybody thinks we shouldn't I'll remove it.